### ISSUE TITLE
MonoBehaviour issue

### ISSUE THREAD

**Issue by mafaca** [DaZombieKiller/TypeTreeTools#10]
This
https://github.com/DaZombieKiller/TypeTreeTools/blob/427d779d68f7a4bce01c2e4339e972e039c9895d/Assembly-TypeTreeTools/Unity/Core/NativeObject.cs#L58
or this
https://github.com/DaZombieKiller/TypeTreeTools/blob/08b360291e32fefee8f6e53e0c9c0e4b63b046b2/Assembly-TypeTreeTools/TypeTreeTools/ExportCommands.Legacy.cs#L157
I guess it is incorrect return value usage. For MonoBehaviour it return false so it doesn't build hierarchy but it can and should (previous TryGetTypeTree implementation did that).
I guess it return false for some other classes too

**Comment by DaZombieKiller** [comment]
Yeah, the return value is rather mysterious. I'm not quite sure what it's supposed to represent currently.
On that note, this project has been superceded by https://github.com/DaZombieKiller/TypeTreeDumper which is much more versatile.

**Comment by mafaca** [comment]
What is the fundamental difference between those two projects?

**Comment by DaZombieKiller** [comment]
TypeTreeDumper is a rewrite of this project and doesn't run inside Unity itself, so it can make use of newer .NET features to interop with the native parts of the engine more smoothly.
